pkg/scheduler/framework/plugins/podtopologyspread/filtering_test.go

```diff
@@ -818,6 +819,31 @@ func TestPreFilterStateAddPod(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "add a pod that doesn't match the node affinity",
+			preemptor: st.MakePod().Name("p").Label("foo", "").NodeAffinityNotIn("foo", []string{"bar"}).
+				SpreadConstraint(1, "zone", v1.DoNotSchedule, st.MakeLabelSelector().Exists("foo").Obj()).
+				Obj(),
+			nodeIdx:  0,
+			addedPod: st.MakePod().Name("p-a1").Node("node-b").Label("foo", "").Label("zone", "zone2").Obj(),
```

Comment:
nit: p-a1 -> p-b1, zone:zone label is not needed

```suggestion
			addedPod: st.MakePod().Name("p-b1").Node("node-b").Label("foo", "").Obj(),
```